src/server.rs

```diff
@@ -153,11 +160,96 @@ impl Server {
         }
     }
 
+    /// Runs any currently pending callbacks
+    ///
+    /// This runs all currently pending callbacks on the current
+    /// thread.
+    ///
+    /// This should be called frequently (e.g. once per a frame)
+    /// in order to reduce the latency between recieving events.
+    pub fn run_callbacks(&self) {
```

Comment:
Instead of duplicating the code here, it might be a good idea to implement this on `Inner` and just defer the calls from both `Client` and `Server`  to it instead.